docs/howto/operate/ci-cd.md

```diff
@@ -0,0 +1,138 @@
+to get a temporary kubeconfig file that allows a successful connection with the
+kubernetes cluster.
+It is actually expecting a 2i2c generated IAM entity (the `deployer` user) and its
+AWS credentials (that we provide in a sops encrypted json file) to properly set them up
+as the AWS `Access Key ID` and `Secret Access Key` as environment variables and
+successfully perform operations in the cluster as the `deployer` user.
+
+There are some requisites for being able to successfully and automatically deploy hubs
+on AWS clusters using the provided functionality:
+
+1. Know if you are working with a [AWS EKS-based](https://aws.amazon.com/eks/) or a
+[kops-based](https://kops.sigs.k8s.io/getting_started/aws/) cluster
+  
+* We currently deploying `EKS`-based (ie, Carbonplan) and `kops`-based cluster (ie, Farallon).
+Whether to create one or another depends on the cluster operational needs and some other
+factors outlined (and being discussed on [this issue](https://github.com/2i2c-org/pilot-hubs/issues/431)). 
+You need to know and understand which sort of cluster are you working with to properly
+set up the config file in step 4.
+
+Look at [this section](new-cluster:aws) if you are interested in setting up a `kops`-based
+cluster.
+
+:::{note}
+Documentation about creating an `EKS`-based cluster has not yet being written
+:::
+
+2. Create a `deployer` user with enough permissions for being able to automatically
+retrieve the kubernetes context.
+
+:::{note}
+We are currently setting up the `deployer` user with `AdministratorAccess`, we may want
+to restrict the permissions scope in the future to reduce the risk surface.
+:::
+
+* Create a new user
+
+  aws iam create-user --user-name deployer
+
+* Add the user to the `2i2c group` (assuming that group exists and have enough permissions
+to operate the cluster)
+
+  aws iam add-user-to-group --group-name 2i2c --user-name deployer
+
+* Check if the user was added to the group
+  
+  aws iam get-group --group-name 2i2c
+
+:::{note}
+It might be the case you do not have an existing `2i2c group`, that is not a problem as
+long as you have enough permissions at the user level.
+:::
+
+* Check permissions associated with that group/user
+
+  aws iam list-attached-group-policies --group-name 2i2c
+
+  aws iam list-attached-user-policies --user-name deployer
+
+If you are working with a EKS-based cluster, you need to perform these additional steps:
+
+  * Add the user to the EKS cluster
+
+    eksctl create iamidentitymapping --cluster <NAME_OF_THE_CLUSTER> --region=<REGION> --arn arn:aws:iam::<ACCOUNT_ID>:user/deployer --group system:masters --username admin
+
+  * Check if the user was added to the map
+
+    eksctl get iamidentitymapping --cluster <NAME_OF_THE_CLUSTER> --region=<REGION>
+
+3. Generate credentials for such a user and encrypted them
+
+* Get the `deployer`credentials with
+
+  aws iam create-access-key --user-name deployer | tee /tmp/deployer.json
+
+:::{admonition}
```

Comment:
I'd just add this as another bullet-point step, so that a reader would expect to follow it in a linear fashion like any other step in the guide